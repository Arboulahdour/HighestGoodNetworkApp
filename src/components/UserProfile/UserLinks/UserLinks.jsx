import React from 'react'

import { Card, CardTitle, CardText, Badge, Button, CardBody } from 'reactstrap'

const UserLinks = ({
	linkType,
	links = [],
	handleModelState,
	isUserAdmin,
	canEditFields
}) => {
	return (
		<Card body>
			<CardTitle style={{ fontWeight: 'bold', fontSize: 20 }}>
				{linkType} Links :
			</CardTitle>
			<CardBody>
				{!links.length && (
					<CardText>
						<Badge color='danger'>No Links present</Badge>
					</CardText>
				)}
				{links.map((item, key) => (
					<CardText
						key={key}
						style={{
							fontSize: 20,
							justifyContent: 'space-between',
							display: 'flex',
							paddingLeft: 20,
							paddingRight: 20
						}}>
						<Badge style={{ width: '10vw' }} color='secondary'>
							{item.Name}
						</Badge>
						<Badge style={{ width: '40vw' }} href={item.Link} color='warning'>
							{item.Link}
						</Badge>
					</CardText>
				))}
			</CardBody>

			{linkType === 'Admin' && isUserAdmin && (
				<Button
					outline
					color='primary'
					onClick={() => handleModelState(true, 'input', linkType)}>
					Add a new Link
				</Button>
			)}
			{linkType !== 'Admin' && (
				<Button
					outline
					color='primary'
					onClick={() => handleModelState(true, 'input', linkType)}>
					Add a new Link
				</Button>
			)}
		</Card>
	)
}

export default UserLinks
