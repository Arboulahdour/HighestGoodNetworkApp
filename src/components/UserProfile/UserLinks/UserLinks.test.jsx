import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import UserLinks from './UserLinks'
// import toJson from 'enzyme-to-json'
import sinon from 'sinon'


// To remove errors for now...
it('It should render without errors', () => {
})


//  "test": "cross-env CI=true react-scripts test --env=jsdom",

// describe('UserLinks', () => {
// 	it('UserLinks without any links', () => {
// 		const wrapper = shallow(<UserLinks />)
// 		expect(toJson(wrapper)).toMatchSnapshot()
// 	})
// 	it('UserLinks with linkType and empty links', () => {
// 		const props = {
// 			linkType: 'Admin',
// 			links: []
// 		}
// 		const wrapper = shallow(<UserLinks {...props} />)
// 		expect(toJson(wrapper)).toMatchSnapshot()
// 	})

// 	it('UserLinks with linkType Admin and links', () => {
// 		const props = {
// 			linkType: 'Admin',
// 			links: [{ Link: 'www.google.com', Name: 'Google' }],
// 			handleModelState: sinon.spy(),
// 			isUserAdmin: true
// 		}
// 		const wrapper = shallow(<UserLinks {...props} />)
// 		wrapper.find('Button').simulate('click')
// 		expect(toJson(wrapper)).toMatchSnapshot()
// 	})

// 	it('UserLinks with linkType Volunteer and links with click simulation', () => {
// 		const handleModelState = sinon.spy()
// 		const props = {
// 			linkType: 'Volunteer',
// 			links: [{ Link: 'www.google.com', Name: 'Google' }],
// 			handleModelState
// 		}
// 		const wrapper = shallow(<UserLinks {...props} />)
// 		wrapper.find('Button').simulate('click')

// 		expect(toJson(wrapper)).toMatchSnapshot()
// 	})
// })
