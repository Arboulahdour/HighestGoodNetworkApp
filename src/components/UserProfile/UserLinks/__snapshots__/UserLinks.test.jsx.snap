// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLinks UserLinks with linkType Admin and links 1`] = `
<Card
  body={true}
  tag="div"
>
  <CardTitle
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
    tag="div"
  >
    Admin
     Links :
  </CardTitle>
  <CardBody
    tag="div"
  >
    <CardText
      key="0"
      style={
        Object {
          "display": "flex",
          "fontSize": 20,
          "justifyContent": "space-between",
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
      tag="p"
    >
      <Badge
        color="secondary"
        pill={false}
        style={
          Object {
            "width": "10vw",
          }
        }
        tag="span"
      >
        Google
      </Badge>
      <Badge
        color="warning"
        href="www.google.com"
        pill={false}
        style={
          Object {
            "width": "40vw",
          }
        }
        tag="span"
      >
        www.google.com
      </Badge>
    </CardText>
  </CardBody>
  <Button
    color="primary"
    onClick={[Function]}
    outline={true}
    tag="button"
  >
    Add a new Link
  </Button>
</Card>
`;

exports[`UserLinks UserLinks with linkType Volunteer and links with click simulation 1`] = `
<Card
  body={true}
  tag="div"
>
  <CardTitle
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
    tag="div"
  >
    Volunteer
     Links :
  </CardTitle>
  <CardBody
    tag="div"
  >
    <CardText
      key="0"
      style={
        Object {
          "display": "flex",
          "fontSize": 20,
          "justifyContent": "space-between",
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
      tag="p"
    >
      <Badge
        color="secondary"
        pill={false}
        style={
          Object {
            "width": "10vw",
          }
        }
        tag="span"
      >
        Google
      </Badge>
      <Badge
        color="warning"
        href="www.google.com"
        pill={false}
        style={
          Object {
            "width": "40vw",
          }
        }
        tag="span"
      >
        www.google.com
      </Badge>
    </CardText>
  </CardBody>
  <Button
    color="primary"
    onClick={[Function]}
    outline={true}
    tag="button"
  >
    Add a new Link
  </Button>
</Card>
`;

exports[`UserLinks UserLinks with linkType and empty links 1`] = `
<Card
  body={true}
  tag="div"
>
  <CardTitle
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
    tag="div"
  >
    Admin
     Links :
  </CardTitle>
  <CardBody
    tag="div"
  >
    <CardText
      tag="p"
    >
      <Badge
        color="danger"
        pill={false}
        tag="span"
      >
        No Links present
      </Badge>
    </CardText>
  </CardBody>
</Card>
`;

exports[`UserLinks UserLinks without any links 1`] = `
<Card
  body={true}
  tag="div"
>
  <CardTitle
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
    tag="div"
  >
     Links :
  </CardTitle>
  <CardBody
    tag="div"
  >
    <CardText
      tag="p"
    >
      <Badge
        color="danger"
        pill={false}
        tag="span"
      >
        No Links present
      </Badge>
    </CardText>
  </CardBody>
  <Button
    color="primary"
    onClick={[Function]}
    outline={true}
    tag="button"
  >
    Add a new Link
  </Button>
</Card>
`;
